### DISCUSSION TITLE
[v.0.5.0] Test and play with the improved first start-up procedure

### DISCUSSION THREAD

**Discussion by radio24** [radio24/TorBox#111] [Archive]
With [TorBox v.0.4.2](https://github.com/radio24/TorBox/releases/tag/v.0.4.2-02.08.2021), the install scripts were extensively rewritten to be more reliable and secure in highly authoritarian countries. Tor will only be activated after a restart and a login by SSH or a web browser. This allows the user to start pluggable transports and bridges from the beginning to hide the use of tor in a better way and improve the user's security. 

Unfortunately, some users struggled to get TorBox working because they activated OBFS4 in an environment with a tightly configured firewall. Also, we like to allow users to start TorBox's automatic counteractions (TACA) and/or Vanguards but doesn't want to impose it as default. This is the reason we improved the start-up procedure, which will be released with the next TorBox version between the end of this month and the beginning of next month.

However, you can already test and play with the improved first start-up and help us make it even better with your feedback. Also, this may help people who have problems starting TorBox.
&nbsp;
### How to install and activate the improved first start-up procedure?

Install the TorBox menu from the v.0.4.3 test branch. Go to the TorBox command prompt by pressing ESC in the TorBox Main menu and use the following commands:
```
cd
wget https://github.com/radio24/TorBox/archive/refs/heads/v.0.4.3.zip
sudo rm -r torbox
unzip v.0.4.3.zip
mv TorBox-v.0.4.3 torbox
rm v.0.4.3.zip
cd torbox
sudo cp /etc/tor/torrc /etc/tor/torrc.bak
sudo cp etc/tor/torrc /etc/tor/
```

Activate and start the first start-up procedure:
```
sudo sed -i "s/^FRESH_INSTALLED=.*/FRESH_INSTALLED=3/" run/torbox.run
./menu
```
&nbsp;
### First start-up manual
The first start-up procedure will ask you the following:
- from where the Internet is coming;
- if you want to use OBFS4 bridges;
- if you want to activate the TorBox automatic counter actions and/or Vanguards.

&nbsp;
**Step 1: Welcome Screen**

![Welcome Screen](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-001-e1630760281227.png)

&nbsp;
**Step 2: From where the Internet is coming**

![From where the Internet is coming](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-002-e1630760405196.png)
&nbsp;
Depending on which Internet source is selected, further questions are asked, such as whether or not TorBox is behind a captive portal (see image below). The code behind this routine is the same as the Main Menu used to select and/or change the Internet source.
&nbsp;
![Is TorBox behind a Captive Portal?](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-003-e1630761290368.png)

&nbsp;
**Step 3: Activating TACA and/or Vanguards**

![Activating TACA and/or Vanguards](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-004-e1630761338826.png)

&nbsp;
**Step 4: Activating OBFS4 bridges (only recommended if necessary)**

![Activating TACA and/or Vanguards](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-005-e1630762136590.png)

Activating OBFS4 can take up to 5 minutes without showing new log entries. Just be patient. Also, there may be error messages - don't freak out and just wait until tor is started up. If successful, you should see "Bootstrapped 100%: Done". After pressing CTRL-C, TorBox will ask you if the bootstrap was successful (see image below).

![Did you see "Bootstrapped 100%: Done" [Y/n]?](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-006.png)

&nbsp;
If the answer is no, TorBox gives you the possibility to circumvent tightly configured firewalls.

![Activating TACA and/or Vanguards](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-007-e1630762922493.png)

&nbsp;
After restarting tor, TorBox will ask again if the bootstrap was successful. In case of a repeated failure, TorBox deactivates the anti-firewall countermeasure and starts the Countermeasure sub-menu, where you can deactivate OBFS4 bridges or try another pluggable transport (MEEK-AZURE/SNOWFLAKE).

![Activating TACA and/or Vanguards](https://www.torbox.ch/wp-content/uploads/2021/09/first_use-008-e1630763182131.png)

**Comment by nyxnor** [comment]
Tested :100:
